### PR TITLE
Entity format

### PR DESCRIPTION
**Pull request by tyt2y3**
#105

### CODE REVIEW

**Comment by Acidic9** [comment]
Looking awesome!

We don't want to use the shorter `#[sea(...)]` instead of `#[sea_orm(...)]`?

Also, if we wanted this to be more correct (something I even forgot to do), we should add `::` before the types. For example `impl ::std::default::Default ... ` and `impl ::sea_orm::...`, but it's not super important.

Do you think it's worth updating all the macros to take in an `DeriveInput` parameter and return `syn::Result<TokenStream>`, and use the struct to store the data?
Now it has a few inconsistencies, and updating it like that would allow a macro_rules for simplifying the lib.rs file.
Eg:
```rust
macro_rules! create_derive {
    ($name:ident $(, $( $attrs:ident ),+)?) => {
        paste! {
            #[proc_macro_derive($name, attributes($( $( $attrs ),*)?))]
            pub fn [< $name:snake >](input: TokenStream) -> TokenStream {
                let input = parse_macro_input!(input as DeriveInput);
                derives::[<expand_ $name:snake>](input)
                    .unwrap_or_else(Error::into_compile_error)
                    .into()
            }
        }
    }
}

create_derive!(DeriveEntity, sea);
create_derive!(DerivePrimaryKey);
create_derive!(DeriveColumn);
create_derive!(DeriveCustomColumn);
create_derive!(DeriveModel, sea);
create_derive!(DeriveModelColumn, sea);
create_derive!(DeriveModelPrimaryKey, sea);
create_derive!(DeriveActiveModel);
create_derive!(DeriveActiveModelBehavior);
create_derive!(DeriveRelation, sea);
create_derive!(FromQueryResult);
```
*(the macro above works)*

**Comment by tyt2y3** [comment]
> We don't want to use the shorter `#[sea(...)]` instead of `#[sea_orm(...)]`?
> 
I guess the norm is to use the crate's name itself

> Also, if we wanted this to be more correct (something I even forgot to do), we should add `::` before the types. For example `impl ::std::default::Default ... ` and `impl ::sea_orm::...`, but it's not super important.
> 
I think it's fine if someone wants to alias something else as sea_orm.

> Do you think it's worth updating all the macros to take in an `DeriveInput` parameter and return `syn::Result<TokenStream>`, and use the struct to store the data?
> 
> Now it has a few inconsistencies, and updating it like that would allow a macro_rules for simplifying the lib.rs file.

Yes I do think it's a good idea.

**Comment by Acidic9** [comment]
> I guess the norm is to use the crate's name itself?

I think it's fine having a shorter version, nothing will conflict with `sea`. For example, the async-graphql crate uses `#[graphql(...)]`, instead of the full `async-graphql`.

> I think it's fine if someone wants to alias something else as sea_orm.

Sure, it's a very small thing. But I think it's usually a standard thing when writing macros, mainly for `::std::...` std library types.

**Comment by tyt2y3** [comment]
I hope `sea_orm` is not too long to be obstructive )
I do agree on the `::std` thing, but we actually `use create as sea_orm` ourselves
https://github.com/SeaQL/sea-orm/blob/a186338ff5794553bdd0b53fdcb4310deeece0a5/src/tests_cfg/fruit.rs#L1

File: sea-orm-macros/src/derives/entity_model.rs [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+            match self {
+                #columns_trait
+            }
+        }
+    }
+
+    #entity_def
+
+    #primary_key
+        });
```
Comment (billy1624) [inline sea-orm-macros/src/derives/entity_model.rs:246]: Btw... there are some clippy warning here loll

**Review by billy1624 — CHANGES_REQUESTED** [review]

**Review by billy1624 — APPROVED** [review]

**Pull request merged**